File: 2026-05/2026-05-27/pizza_party.py
import math
from functools import reduce

def get_pizzas_to_order(hours_worked):
    HOURS_WORKED_PER_SLICE = 3
    MINIMUM_SLICES_PER_PERSON = 2
    SLICES_PER_PIZZA = 8

    def total_slice_count (total, hours):
        person_s_slice_count = math.ceil(hours / HOURS_WORKED_PER_SLICE)

        return total + (2 if person_s_slice_count < 2 else person_s_slice_count)

    return math.ceil(reduce(total_slice_count, hours_worked, 0) / 8)

# --- TEST SUITE ---

tests_text = r'''
1. get_pizzas_to_order([8, 8, 8]) should return 2.
2. get_pizzas_to_order([10, 9, 8, 2, 2, 6, 10]) should return 3.
3. get_pizzas_to_order([1, 2, 3, 4, 5]) should return 2.
4. get_pizzas_to_order([8, 8, 8, 8, 8, 8, 8, 8]) should return 3.
5. get_pizzas_to_order([9, 9, 6]) should return 1.
6. get_pizzas_to_order([10, 12, 16, 9, 8, 11, 15, 8, 0]) should return 5.
'''

import re

tests_regex = re.compile(r"(?P<number>\d+)\.\s(?P<function_call>.+) should return (?P<output>.+?)\.?$", re.MULTILINE)
test_data = [item.groupdict() for item in re.finditer(tests_regex, tests_text)]

def run_tests(test_data):
    print("——————————————————————————",
        "\n🧪 Starting Verification...",
        "\n——————————————————————————")

    fail_count = 0

    for test in test_data:
        function_call_output = eval(test['function_call'])
        test_output = eval(test['output'])
        
        if function_call_output == test_output:
            print(f"{test['number']}.✅ PASS - Function Call:\n{test['function_call']}")
        else:
            print(f"{test['number']}.❌ FAIL - Function Call:\n{test['function_call']}\nExpected:\n{test_output}\nGot:\n{function_call_output}")
            fail_count += 1

        print("————————————————————————————")

    print(
        f"⚠️ WARNING: {fail_count}/{len(test_data)} tests FAILED." if fail_count
        else "🎉 SUCCESS: All tests PASSED."
    )

run_tests(test_data)

File: 05 May/05-27/pizzaParty.js
function getPizzasToOrder(hoursWorked) {
  const HOURS_WORKED_PER_SLICE = 3;
  const MINIMUM_SLICES_PER_PERSON = 2;
  const SLICES_PER_PIZZA = 8;

  return Math.ceil(hoursWorked.reduce((acc, hours) => {
    const personSSliceCount = Math.ceil(hours / HOURS_WORKED_PER_SLICE);

    return acc + (personSSliceCount < MINIMUM_SLICES_PER_PERSON ? MINIMUM_SLICES_PER_PERSON : personSSliceCount);
  }, 0) / SLICES_PER_PIZZA);
}

// --- TEST SUITE ---

const testsText = String.raw`
1. getPizzasToOrder([8, 8, 8]) should return 2.
2. getPizzasToOrder([10, 9, 8, 2, 2, 6, 10]) should return 3.
3. getPizzasToOrder([1, 2, 3, 4, 5]) should return 2.
4. getPizzasToOrder([8, 8, 8, 8, 8, 8, 8, 8]) should return 3.
5. getPizzasToOrder([9, 9, 6]) should return 1.
6. getPizzasToOrder([10, 12, 16, 9, 8, 11, 15, 8, 0]) should return 5.
`;

const testsRegex = /(?<number>\d+)\.\s(?<functionCall>.+) should return (?<output>.+?)\.?$/gm;
const testData = [...testsText.matchAll(testsRegex).map(match => match.groups)];

function runTests(testData) {
  console.log("——————————————————————————",
            "\n🧪Starting Verification...",
            "\n——————————————————————————");

  let failCount = 0;
  
  testData.forEach(test => {
    const functionCallOutput = eval(test.functionCall);
    const testOutput = eval(`(${test.output})`);

    if (JSON.stringify(functionCallOutput) === JSON.stringify(testOutput)) {
      console.log(`${test.number}.✅PASS - Function Call: ${test.functionCall}`);
    } else {
      console.log(`${test.number}.❌FAIL - Function Call: ${test.functionCall}\nExpected: ${JSON.stringify(testOutput)}\nGot: ${JSON.stringify(functionCallOutput)}`);

      ++failCount;
    }
    console.log("————————————————————————————");
  })

  console.log(failCount
    ? `⚠️WARNING: ${failCount}/${testData.length} tests FAILED.`
    : "🎉SUCCESS: All tests PASSED."
  );
}

runTests(testData);